File: Leetcode/Daily/07-2025/Java/05.java
// Problem: 1394. Find Lucky Integer in an Array (Easy)
// Link: https://leetcode.com/problems/find-lucky-integer-in-an-array

class Solution {
    public int findLucky(int[] arr) {
        HashMap<Integer, Integer> umap = new HashMap<>();
        
        for (int i: arr) umap.put(i, umap.getOrDefault(i, 0) + 1);
        
        int ans = -1;
        for (int key: umap.keySet()) {
            if (key == umap.get(key)) ans = Math.max(ans, key);
        }
        
        return ans;        
    }
}

File: Leetcode/Daily/07-2025/TypeScript/05.ts
// Problem: 1394. Find Lucky Integer in an Array (Easy)
// Link: https://leetcode.com/problems/find-lucky-integer-in-an-array

function findLucky(arr: number[]): number {
    const umap = new Map<number, number>();
    
    for (const i of arr) umap.set(i, (umap.get(i) || 0) + 1);
    
    let ans = -1;
    for (const [key, value] of umap.entries()) {
        if (key === value) ans = Math.max(ans, key);
    }
    
    return ans;
};